sda-commons-server-key-mgmt/README.md

```diff
@@ -0,0 +1,76 @@
+# sda-commons-server-key-mgmt
+Bundle for key management in microservices.
+
+The main purpose is to provide configurable key management and mappings for keys and its values.
+This allows to define keys and mappings at deployment time and not at development time. 
+So the actual keys can be adjusted according to the deployment scenario. 
+
+When defining keys at runtime, it might happen that keys used in business logic do not exist at runtime. 
+This problem is not solved by the bundle. 
+It must be considered as part of the key definition process. 
+
+The bundle provide means to work with keys in  to retrieve mapping values from or to a possible implementation depending on the keys.
+This includes
+  * checking of a key is valid
+  * checking if a value is valid for a given key
+  * mapping of a key between API and implementation specific values
+
+API keys and values should be defined in snake case. For these two values, the bundle will be case tolerant, by mapping to uppercase internally.
+For implementations specific values, the bundle must not be case tolerant to provide keys as expected.
```

Comment:
Could you add a clear explanation about the desired purpose of API and implementation? Maybe also in the JavaDoc. As both can be configured very flexible, it is not 100% clear to me which values should be used in which places.

Apart from this, I'm fine with the PR, but I'd like @christopher-cudennec to have a look as well.